migrator/src/main/scala/com/scylladb/migrator/DynamoUtils.scala

```diff
@@ -50,19 +51,31 @@ object DynamoUtils {
           .withTableName(target.table)
           .withKeySchema(sourceDescription.getKeySchema)
           .withAttributeDefinitions(sourceDescription.getAttributeDefinitions)
-          .withProvisionedThroughput(
-            new ProvisionedThroughput(
-              sourceDescription.getProvisionedThroughput.getReadCapacityUnits,
-              sourceDescription.getProvisionedThroughput.getWriteCapacityUnits
+        if (sourceDescription.getProvisionedThroughput.getReadCapacityUnits != 0L && sourceDescription.getProvisionedThroughput.getWriteCapacityUnits != 0) {
+          request
+            .setProvisionedThroughput(
+              new ProvisionedThroughput(
+                sourceDescription.getProvisionedThroughput.getReadCapacityUnits,
+                sourceDescription.getProvisionedThroughput.getWriteCapacityUnits
+              )
             )
-          )
+        } else {
+          request.setBillingMode(BillingMode.PAY_PER_REQUEST.toString)
+        }
 
         log.info(
           s"Table ${target.table} does not exist at destination - creating it according to definition:")
         log.info(sourceDescription.toString)
         targetClient.createTable(request)
         log.info(s"Table ${target.table} created.")
 
+        var targetTableDesc = targetClient.describeTable(target.table).getTable
+        while (TableStatus.ACTIVE.toString != targetTableDesc.getTableStatus) {
+          log.debug(s"Target table not ready yet. Waiting 1 second.")
+          Thread.sleep(1000)
```

Comment:
so there should be a timeout to this ... e.g. 
5 mins worth of attempts?

or 2mins?

Reply:
of course fail nicely if timeout is hit explaining what broke

Reply:
We use `sleep` already when we enable the streams:

https://github.com/scylladb/scylla-migrator/blob/f24cfa2786f0edd5498e10c05e2bdfa6197058f1/migrator/src/main/scala/com/scylladb/migrator/DynamoUtils.scala#L113

If we upgrade the AWS SDK to the v2 we can use the new [Waiter](https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/waiters.html) abstraction.

Reply:
okay, filing the issue then, let's target sdk2